File: jHyuk/week11/요격 시스템.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(vector<int>& v1, vector<int>& v2) {
    if (v1[1] == v2[1])
        return v1[0] < v2[0];
    else return v1[1] < v2[1];
}

int solution(vector<vector<int>> targets) {
    int answer = 0;
    int idx = 0;
    sort(targets.begin(), targets.end(), cmp);

    for (int i = 0; i < targets.size(); i++) {
        if (targets[i][0] < idx) {
            continue;
        }
        else {
            idx = targets[i][1];
            answer++;
        }

    }

    return answer;
}

File: lujae/11주차/요격시스템.py
# 시간 : 10분 51초
# 레벨 : 2

targets = [[4,5],[4,8],[10,14],[11,13],[5,12],[3,7],[1,4]]

targets.sort(key = lambda x: x[1])

mainObj = targets[0]

answer = 1
for i in range(1, len(targets)):
    currentObj = targets[i]

    if currentObj[0] < mainObj[1]:
        continue

    mainObj = currentObj
    answer += 1

print(answer)